server/events/event_parser.go

```diff
@@ -343,11 +346,18 @@ func (e *EventParser) ParseAPIPlanRequest(vcsHostType models.VCSHostType, repoFu
 
 // GetBitbucketCloudPullEventType returns the type of the pull request
 // event given the Bitbucket Cloud header.
-func (e *EventParser) GetBitbucketCloudPullEventType(eventTypeHeader string) models.PullRequestEventType {
+func (e *EventParser) GetBitbucketCloudPullEventType(eventTypeHeader string, sha string, pr string) models.PullRequestEventType {
 	switch eventTypeHeader {
 	case bitbucketcloud.PullCreatedHeader:
+		lastBitbucketSha.Add(pr, sha)
 		return models.OpenedPullEvent
 	case bitbucketcloud.PullUpdatedHeader:
+		lastSha, _ := lastBitbucketSha.Get(pr)
+		if sha == lastSha {
+			// No change, ignore
+			return models.OtherPullEvent
+		}
+		lastBitbucketSha.Add(pr, sha)
```

Comment:
Is there a better way to solve this problem than an LRU cache?

What do you folks think @runatlantis/maintainers ?

Reply:
I commented on the issue about this, I'm not fond of the LRU unless absolutely necessary. We could update the PR model to track previous SHA in the db state. That way we are consistent with the approach across VCS backends.

However, most VCS backends solve this with a proper event that lets you know when the source branch is updated. Bitbucket Cloud doesn't have this.

Reply:
@GenPage  To clarify, the LRU is not ideal but it's fine for now? I'm wondering if this blocks the PR merge.

Reply:
@GenPage following up on the above